app/controllers/content_items_controller.rb

```diff
@@ -137,17 +137,55 @@ def set_guide_draft_access_token
   end
 
   def load_content_item
+    @content_item = if use_graphql?
+                      schema_name_result = Services
+                                             .publishing_api
+                                             .graphql_content_item(Graphql::SchemaNameQuery.new(content_item_path).query)
+
+                      if schema_name_result["schema_name"] == "news_article"
+                        load_content_item_from_graphql
+                      else
+                        load_content_item_from_content_store
+                      end
+                    else
+                      load_content_item_from_content_store
+                    end
```

Comment:
**thought:** two requests issue

I suppose if we were to switch over to GraphQL for everything, we might be able to remove the schema name query and do what I suggested in the meeting with Richard and Martyn: have a single big query that specifies what we want for each type using the `...Type` syntax. We'd have to weigh up whether one big query is better than one very small one and then the normal schema-specific one

I think this could work with both the current more strongly typed implementation and the Edition god type alternative. With the latter, we could have types that don't themselves define any fields but do at least map document types to GraphQL types, like you've already done for news articles: https://github.com/alphagov/publishing-api/blob/main/app/graphql/types/news_article_type.rb. Then whichever `...Type` section of the query matches would be executed

Reply:
Would it be gross to change this to just be 
``` 
 @content_item = use_graphql? ? load_content_item_from_graphql : load_content_item_from_content_store
 ```
 and then at the top of the `load_content_item_from_graphql` method we call a new method that checks for the content type and either carries on using graphql or just returns `load_content_item_from_content_store`
 
 I know that it hides a bit of functionality but it would make this bit of code look nicer

Reply:
I've reduced this down to one GraphQL query now.  We'll need to think about the query a bit more once any changes to the typing in Publishing API have been made and when we add GraphQL support for further document types rendered by Government Frontend.